src/components/Map.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from 'react';
+import GoogleMapReact from 'google-map-react';
+
+import withResource from './withResource';
+import { TResource } from '../types';
+
+type Props = { data: TResource };
+
+const boulderCoordinates = {
+  lat: 40.0156852,
+  lng: -105.2792069
+};
+
+class Map extends Component<any, any> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      directionsAreToggled: false
+    };
+  }
+
+  calculateAndDisplayRoute = (
+    directionsService: any,
+    directionsDisplay: any,
+    resourceAsDestination: any
+  ) => {
+    const userLocation = this.props.currentUser.location;
+
+    // remove default markers A and B (origin/destination)
+    directionsDisplay.setOptions({ suppressMarkers: true });
+
+    directionsService.route(
+      {
+        origin: userLocation,
+        destination: {
+          lat: resourceAsDestination.lat,
+          lng: resourceAsDestination.lng
+        },
+        travelMode: 'TRANSIT'
+      },
+      function(response: any, status: any) {
+        if (status === 'OK') {
+          directionsDisplay.setDirections(response);
+        } else {
+          console.log('Directions request failed due to ' + status);
```

Comment:
```suggestion
          console.log(`Directions request failed due to ${status}`);
```
Just a personal preference on this 😛